src/process-mining-visualisation/mining-action.ts

```diff
@@ -52,16 +53,17 @@ export class MiningCommand extends Command {
     super();
   }
 
+  @inject(EdgeRouterRegistry) edgeRouterRegistry: EdgeRouterRegistry;
+
   execute(context: CommandExecutionContext): CommandReturn {
     const model = context.root;
     this.action.data.nodes.forEach(node => {
-      const element = model.index.getById(node.ID);
-      if (element instanceof Edge) {
-        if (element.args === undefined) {
-          element.args = {};
-        }
-        element.args['labelvalue'] = node.labelvalue;
-        element.args['relativevalue'] = node.relativevalue;
+      const edge = model.index.getById(node.ID);
+      if (edge instanceof Edge) {
+        const segments = this.edgeRouterRegistry.route(edge, edge.args);
+        const miningLabel = new MiningLabel(node.labelvalue, node.relativevalue, segments);
+        this.moveExistingLabel(edge.editableLabel as GLabel, segments);
```

Comment:
why do you need to move the existing label?

Reply:
I move the label to try and avoid overlapping:
![image](https://github.com/user-attachments/assets/ce9dc43f-cf35-495e-a423-809d94736ea3)
vs
![image](https://github.com/user-attachments/assets/04b9afe8-8f01-4d18-bc5c-00088406dd0f)

but I changed it to only move if the distance between label and mining-annotation to small